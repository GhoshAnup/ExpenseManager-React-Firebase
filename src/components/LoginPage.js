import React from 'react';
import {connect} from 'react-redux';
import { startLogin } from '../actions/auth';
//import { firebase, googleAuthProvider } from '../firebase/firebase';

export const LoginPage = ({startLogin}) => (   
    <div>
      <button onClick={startLogin}>Login</button>
    </div>
  );
  const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
  });  
  export default connect(undefined, mapDispatchToProps)(LoginPage);

// export const LoginPage = () => (
//     <div>
//       <button onClick={startLogin}>Login</button>
//     </div>
//   );

//   const startLogin = () => {
//     firebase.auth().signInWithPopup(googleAuthProvider);
// }
// export default LoginPage;
