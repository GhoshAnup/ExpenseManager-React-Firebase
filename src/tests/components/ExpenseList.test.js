import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('test render expense list with data', () =>{
    const wrapper = shallow(<ExpenseList expenses = {expenses}/>)
    expect(wrapper).toMatchSnapshot();
});

test('test render expense list with no data', () =>{
    const wrapper = shallow(<ExpenseList expenses = {[]}/>) 
    expect(wrapper).toMatchSnapshot();
});