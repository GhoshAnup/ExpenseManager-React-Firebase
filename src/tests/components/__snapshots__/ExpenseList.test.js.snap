// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render expense list with data 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={18934}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={9800}
    createdAt={345600000}
    description="Debit card"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`test render expense list with no data 1`] = `
<div>
  <p>
    no expenses
  </p>
</div>
`;
